app/javascript/dashboard/helper/ReconnectService.js

```diff
@@ -0,0 +1,130 @@
+import { BUS_EVENTS } from 'shared/constants/busEvents';
+import { differenceInSeconds } from 'date-fns';
+import {
+  isAConversationRoute,
+  isAInboxViewRoute,
+  isNotificationRoute,
+} from 'dashboard/helper/routeHelpers';
+
+class ReconnectService {
+  constructor(store, router) {
+    this.store = store;
+    this.router = router;
+    this.disconnectTime = null;
+
+    this.setupEventListeners();
+  }
+
+  disconnect = () => this.removeEventListeners();
+
+  getSecondsSinceDisconnect = () =>
+    this.disconnectTime
+      ? Math.max(differenceInSeconds(new Date(), this.disconnectTime), 0)
+      : 0;
+
+  handleOnlineEvent = () => {
+    if (this.getSecondsSinceDisconnect() >= 10800) window.location.reload(); // Force reload if the user is disconnected for more than 3 hours
```

Comment:
Replace the hardcoded magic number 10800 seconds (3 hours) with a named constant.

```
const MAX_DISCONNECT_SECONDS = 10800;
handleOnlineEvent = () => {
    if (this.getSecondsSinceDisconnect() >= MAX_DISCONNECT_SECONDS) {
      window.location.reload();
    }
  };
```